src/Adapter/Image/ProductImageFileValidator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Image;
+
+use ImageManager;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\MemoryLimitException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageSizeException;
+
+class ProductImageFileValidator extends ImageValidator
+{
+    private const MEGABYTE_IN_BYTES = 1000000;
```

Comment:
I always wonder which is the right convention nowadays, is 1kb = 1024b or 1kb = 1000b ?

Reply:
1000

Reply:
or not? :D

Reply:
maybe 1024 will be better - https://cseducators.stackexchange.com/questions/4425/should-i-teach-that-1-kb-1024-bytes-or-1000-bytes

Reply:
btw it should not be placed here :)

Reply:
@lartist already worked on this topic in this PR https://github.com/PrestaShop/PrestaShop/pull/29883 he implemented a tooling class to handle this kind of conversion, although the implementation was not validated yet the idea is interesting

I don't know why the initial PR was closed by the way @lartist ? Is it because of a lack of reviews?

Reply:
It was dying :(

Reply:
| @FabienPapet `btw it should not be placed here`

Could you please suggest something following your comments (make them more explicit)? Else everytime we loose some time asking - where do you think I could place it now, so it doesn't at least block the PR? or trying to guess what you meant :smile: (no bad feelings :v: )

Btw I have no where could we put this kind of constant (I added this constant just to avoid magic numbers hanging in code :man_shrugging:)